### ISSUE TITLE
tpf.interact

### ISSUE THREAD

**Issue by gavinramsay** [lightkurve/lightkurve#1014]
#### Example

```python
import lightkurve
# insert code here ...
```

#### Environment

-  platform (e.g. Linux, OSX, Windows):
-  lightkurve version (e.g. 1.0b6):
-  installation method (e.g. pip, conda, source):

**Comment by gavinramsay** [comment]
This is me not being a pythion pro ... 
im trying to use tpf.interact

ive downloaded a tess TFP. 
>>> import lightkurve as lk
>>> import matplotlib.pyplot as plt
>>> tpf = lk.search_targetpixelfile('TIC 1403938').download()
>>> tpf.interact()

however, here is says:

In: 

and expects an input ... but ive no idea what to do? this is me just being stupid ... 
any help appreciated. i'm running lightkurve 2.0.4
thanks
Gav

**Comment by barentsen** [comment]
Thank you for opening an issue @gavinramsay! :+1:

Could it be that you are trying to run `tpf.interact()` inside a standard python console?  (i.e. in a terminal rather than in a browser?)

The `tpf.interact()` feature will only work in a Jupyter Notebook running in a browser.

There should have been a friendly error message explaining this when it is accidentally used in a terminal.  It's a bug that this message didn't show! (/cc @gully) 

Does this make sense?

**Comment by gavinramsay** [comment]
ah ... that does make sense. ive never actually used a python notebook!

**Comment by gully** [comment]
Yes, unfortunately interact only works in Jupyter Notebooks at the moment, and agreed ideally it would inform the user about this limitation.

I just attempted---in `lightkurve` v2.0.4--- to run `tpf.interact()` in IPython on the terminal, and I got no error message or warning.  

`tpf.animate()` checks to see if the command is called from IPython [here](https://github.com/lightkurve/lightkurve/blob/0cd9c2585f80755767fc1b1fe95a40f6424ed755/src/lightkurve/targetpixelfile.py#L1214).  It looks like we do not currently enforce any checks for front-end before attempting to show the interact user interface.  Here is a [stack overflow question](https://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook) with some answers from astropy developers on how to do that.  So a solution here could be to add this sort of checking before interact (and interact_sky, and interact_bls), to ensure that the code is run from inside Jupyter.

**Comment by gavinramsay** [comment]
ok .. thanks! i should really try and use python notebooks:)

**Comment by orionlee** [comment]
The API doc does say "Display an interactive Jupyter Notebook widget to inspect the pixel data."

https://docs.lightkurve.org/reference/api/lightkurve.KeplerTargetPixelFile.interact.html

`interact_sky()` has similar phrasing.

Does it need to be phrased differently?

It sounds like what might help users is to emit a warning (or throw exception) to users when saying the feature only works in Jupyter notebook.

If we go down the path of emitting warning / exception
1. One might need to be careful on what to do for Jupyter environments that don't support bokeh, e.g., Visual Studio Code, google colab, etc.
2. Emitting warning is probably safer than throwing exception. 
   a. What's more, the `interact()` logic is partially tested in regression test, which runs outside of Jupyter too. So throwing exception would complicate the testing too.
